data/helpers.d/postgresql

```diff
@@ -282,11 +282,11 @@ ynh_psql_test_if_first_run() {
 
 		sudo --login --user=postgres psql -c"ALTER user postgres WITH PASSWORD '$psql_root_password'" postgres
 
-		# force all user to connect to local database using passwords
+		# force all user to connect to local databases using hashed passwords
 		# https://www.postgresql.org/docs/current/static/auth-pg-hba-conf.html#EXAMPLE-PG-HBA.CONF
 		# Note: we can't use peer since YunoHost create users with nologin
 		#  See: https://github.com/YunoHost/yunohost/blob/unstable/data/helpers.d/user
-		ynh_replace_string --match_string="local\(\s*\)all\(\s*\)all\(\s*\)peer" --replace_string="local\1all\2all\3password" --target_file="$pg_hba"
+		ynh_replace_string --match_string="local\(\s*\)all\(\s*\)all\(\s*\)peer" --replace_string="local\1all\2all\3md5" --target_file="$pg_hba"
```

Comment:
Hm so this will happen for instances installing postgresql for the first time, but what about existing instances ? Is this just a matter of replacing a line in a conf file ?

If so we can easily write a migration for existing instance using our migration mechanism. If you want to do it, there's a simple one here : https://github.com/YunoHost/yunohost/blob/stretch-unstable/src/yunohost/data_migrations/0001_change_cert_group_to_sslcert.py
(or we can take care of it if you don't feel confident writing / testing it ;))

Reply:
I created a new file `src/yunohost/data_migrations/0011_postgresql_password_to_md5_authentication.py`.
Does it do what you had in mind?

Reply:
Yes that looks very much like it !

I will spin some tests in the next few days once I'm done with boring work stuff from real life